tools/tiller-hack/tiller-hack.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package main provides a tool tiller-hack, this tools allows exclude further
+// tracking of CRDs objects by helm's tiller after they were deployed by means of
+// helm's charts.
+//
+// ./tiller-hack --kubeconfig={location of kube config file} \
+//               --configmap={the name of tiller's config map for istio deployment}
+//               --namespace={the namespace of configmap and tiller,
+//                            by default it is kube-system}
+//               --restart-tiller={true/false, false is default, true will
+//                                 force the tool to restart tiller}
+//
+package main
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/retry"
+
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/golang/protobuf/proto"
+	"k8s.io/helm/pkg/proto/hapi/chart"
+	rspb "k8s.io/helm/pkg/proto/hapi/release"
+)
+
+var (
+	kubeconfig = flag.String("kubeconfig", "",
+		"Absolute path to the kubeconfig file, must be specified.")
```

Comment:
why must this be specified?  I understand you don't want to have multiple calls to the kube api server, but there is already a library in Istio to handle this which works in cluster and out of cluster automatically.  Two calls instead of one for a tool that is called a few times every 3 months at maximum is not that significant.

Reply:
I do not see any value of replacing this func, it does what it is supposed to and does it well. It also work for in-cluster or externally. I do not see any real technical reason to replace it.